mods/simple_service.js

```diff
@@ -19,74 +18,308 @@ function SimpleService(options) {
     this.exports = this.processSpec(this.spec);
 }
 
+/**
+ * Processes an on_setup property and creates request templates
+ * for each resource
+ *
+ * @param {object} conf an endpoint configuration object
+ * @returns {Array} array of request remplates that must be run on service startup
+ */
+function processResources(conf) {
+    var resources = [];
+    if (conf.on_setup) {
+        if (Array.isArray(conf.on_setup)) {
+            conf.on_setup.forEach(function(resourceSpec) {
+                Object.keys(resourceSpec).forEach(function(requestName) {
+                    var requestSpec = resourceSpec[requestName];
+                    requestSpec.method = requestSpec.method || 'put';
+                    resources.push(new Template(requestSpec));
+                });
+            });
+        } else {
+            throw new Error('Invalid config. on_setup must be an array');
+        }
+    }
+    return resources;
+}
+
+/**
+ * Creates a JS function that verifies property equality
+ *
+ * @param catchDefinition the condition in the format of 'catch' and 'return_if' stanza
+ * @returns {Function} a function that verifies the condition
+ */
+function compileCatchFunction(catchDefinition) {
+    function createCondition(catchCond, option) {
+        if (catchCond === 'status') {
+            if (/^[0-9]+$/.test(option.toString())) {
+                return '(res["' + catchCond + '"] === ' + option + ')';
+            } else {
```

Comment:
Perhaps worth checking that the condition is actually of the form `/^[0-9]xx$/i`?